### PR TITLE
 Add openshift ci job re trigger

### PR DESCRIPTION
**Pull request by rnetser**

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest build-container`: Retest build-container

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ci_jobs_trigger/app.py [hunk 1]
```diff
@@ -3,12 +3,21 @@
 from simple_logger.logger import get_logger
 from flask.logging import default_handler
 
+from ci_jobs_trigger.libs.openshift_ci_re_trigger.openshift_ci_re_trigger import (
```
Comment (myakove) [inline ci_jobs_trigger/app.py:6]: if you put `openshift_ci_re_trigger` code under folder please do the same for `zstream_trigger`

File: ci_jobs_trigger/app.py [hunk 2]
```diff
@@ -17,19 +26,36 @@
...
+            slack_errors_webhook_url=get_config(os_environ=OPENSHIFT_CI_ZSTREAM_TRIGGER_CONFIG_OS_ENV_STR).get(
+                "slack_errors_webhook_url"
+            ),
+        )
+
+
+@APP.route("/openshift_ci_job_re_trigger", methods=["POST"])
+def openshift_ci_job_re_trigger():
+    try:
+        job_triggering = JobTriggering(hook_data=request.json, flask_logger=APP.logger)
```
Comment (myakove) [inline ci_jobs_trigger/app.py:47]: let's keep it unified, please rename `flask_logger` to `logger`

File: ci_jobs_trigger/libs/openshift_ci_re_trigger/openshift_ci_re_trigger.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                    logger=self.logger,
+                )
+
+            raise requests.exceptions.RequestException()
+
+        tests_dict = self.get_testsuites_testcase_from_junit_operator(
+            junit_xml=self.get_tests_from_junit_operator_by_build_id()
+        )
+        if self.is_build_failed_on_setup(tests_dict=tests_dict):
+            prow_job_id = self.trigger_job()
```
Comment (myakove) [inline ci_jobs_trigger/libs/openshift_ci_re_trigger/openshift_ci_re_trigger.py:89]: we need to rename either `trigger_job()` or `self.trigger_job()`, preferred to rename the self one

File: ci_jobs_trigger/libs/openshift_ci_re_trigger/openshift_ci_re_trigger.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+OPENSHIFT_CI_RE_TRIGGER_CONFIG_OS_ENV_STR = "OPENSHIFT_CI_RE_TRIGGER_CONFIG"
+
+
+class JobTriggering:
+    def __init__(self, hook_data, flask_logger):
+        self.logger = flask_logger
+
+        self.log_prefix = f"[{shortuuid.random(length=10)}]"
+        self.hook_data = hook_data
+        self.config = get_config(os_environ=OPENSHIFT_CI_RE_TRIGGER_CONFIG_OS_ENV_STR)
```
Comment (myakove) [inline ci_jobs_trigger/libs/openshift_ci_re_trigger/openshift_ci_re_trigger.py:24]: need to exit if not config.

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
`tests` folder should move under `ci_jobs_trigger` folder.

**Review by myakove — APPROVED** [review]

**Comment by redhat-qe-bot1** [comment]
New container for quay.io/redhat_msi/ci-jobs-trigger:latest published

**Pull request merged**